src/irr/video/COpenGLSpecializedShader.cpp

```diff
@@ -204,14 +204,18 @@ void COpenGLSpecializedShader::buildUniformsList()
             for (size_t i = 0ull; i < top.members.count; ++i) {
                 SMember m = top.members.array[i];
                 m.name = top.name + "." + m.name;
+                if (m.count > 1u)
+                    m.name += "[0]";
```

Comment:
can we precompute these name strings, don't want any alloc calls in `setUniformsImitatingPushConstants`

Reply:
can we actually precompute as much as possible?

Reply:
they're precomputed, it's in `buildUniformList()`